WcaOnRails/app/models/user.rb

```diff
@@ -632,15 +632,33 @@ def banned?
     current_teams.include?(Team.banned)
   end
 
+  def ban_end
+    current_team_members.select(:team == Team.banned).first.end_date
```

Comment:
I understand that you copied it from another place, but then again the code that you copied was just terrible. Perhaps it's not your fault, but touching this part of the code gives us a chance to fix it.

- Using `orm_association.select(some_predicate)` (like is the case here) loads all ORM associations from SQL into memory and _then_ iterates them using the predicate to evaluate which items yield true.
- Using `orm_association.where(field_name: field_value)` is a bit more restrictive because it only allows you to write SQL-esque queries (and not a generic expression that might evaluate to a boolean) but it actually also renders the Ruby code invocation as SQL query and is therefor _much_ more efficient.

Reply:
Something along the lines of `current_team_members.where(team: Team.banned).first` should do.

Reply:
I think that ActiveRecord even has an ORM interface for selecting the first entry using SQL `LIMIT 1` directly via `current_team_members.first(team: Team.banned)`, but I am not 100% sure though

Reply:
I mean I did write the original code too so I guess it is kinda my fault :P

Reply:
https://apidock.com/rails/ActiveRecord/FinderMethods/first does first does only take the limit as an argument